pennylane_lightning_gpu/src/simulator/StateVectorCudaMPI.hpp

```diff
@@ -941,6 +955,364 @@ class StateVectorCudaMPI
         return expect_val;
     }
 
+    /**
+     * @brief Scatter a CSR (Compress Sparse Row) format matrix.
+     *
+     * @tparam index_type Integer type used as indices of the sparse matrix.
+     * @param matrix CSR (Compress Sparse Row) format matrix.
+     * @param root Root rank of the scatter operation.
+     */
+    template <class index_type>
+    auto scatterCSRMatrix(std::vector<CSRMatrix<Precision, index_type>> &matrix,
+                          size_t root) -> CSRMatrix<Precision, index_type> {
+        // Bcast num_rows and num_cols
+        size_t local_num_rows;
+        size_t num_col_blocks;
+        std::vector<size_t> nnzs;
+
+        local_num_rows = size_t{1} << this->getNumLocalQubits();
+        num_col_blocks = mpi_manager_.getSize();
+
+        if (mpi_manager_.getRank() == root) {
+            for (size_t j = 0; j < matrix.size(); j++) {
+                nnzs.push_back(matrix[j].values.size());
+            }
+        }
```

Comment:
```suggestion
        size_t local_num_rows = size_t{1} << this->getNumLocalQubits();
        size_t num_col_blocks = mpi_manager_.getSize();
        std::vector<size_t> nnzs;

        if (mpi_manager_.getRank() == root) {
            nnzs.reserve(matrix.size());
            for (size_t j = 0; j < matrix.size(); j++) {
                nnzs.push_back(matrix[j].values.size());
            }
        }
```
Can we do something like this?

Reply:
Sure!